crates/oxc_linter/src/rules/eslint/sort_vars.rs

```diff
@@ -0,0 +1,226 @@
+use std::cmp::Ordering;
+
+use oxc_ast::{
+    ast::{BindingPatternKind, VariableDeclarator},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Default, Clone)]
+pub struct SortVars {
+    ignore_case: bool,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// When declaring multiple variables within the same block, sorting variable names make it
+    /// easier to find necessary variable easier at a later time.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Unsorted variable declarations can make the code harder to read and maintain.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// var b, a;
+    /// var a, B, c;
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// var a, b, c, d;
+    /// var a, B, d;
+    /// ```
+    SortVars,
+    pedantic,
+    pending
+);
+
+impl Rule for SortVars {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let ignore_case = value
+            .get(0)
+            .and_then(|v| v.get("ignoreCase"))
+            .and_then(serde_json::Value::as_bool)
+            .unwrap_or(false);
+
+        Self { ignore_case }
+    }
+
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::VariableDeclaration(var_decl) = node.kind() else {
+            return;
+        };
+
+        if var_decl.declarations.len() <= 1 {
+            return;
+        };
+
+        let binding_declarations: Vec<&VariableDeclarator> = var_decl
```

Comment:
Can we iterate without collecting?

Reply:
We can, I grabbed some inspiration from sort_imports. It still uses filter() in theory that can be refactored out as well.